### DISCUSSION TITLE
How to make an atom's default value relies on RecoilValue's evaluated value

### DISCUSSION THREAD

**Discussion by wendyzhaogogo** [facebookexperimental/Recoil#1711] [Q&A] [answered]
Motivation Case:
https://codesandbox.io/s/need-atom-default-to-be-a-simple-value-ljqfcm?file=/src/index.tsx
In this case,the page is splited into  three parts:
1.  `date`,`select` is to gather params asynchronously or synchronously.
2.  `search` button is to ensure params to be submitted, only when params are submitted (`search` is clicked), searching  should start.
3.  `table` is to display searching result, in my case ,I only display my params to show the change of it.
4.   On initialization, submitted params will be auto evaluated once to trigger searching once
5.   the immediate change of `date`  `select`  should not trigger searching immediately until I've clicked `search`

Problem:
1. In this case, if `search` has not been clicked , the immediate change of `date` `select`  will trigger searching immediately, it makes wrong.
2. Summarizing to Problem 1, if a selector is assigned to an atom's default value, then the change of the selector's dependencies will trigger the evaluation of atom's downstream nodes

Expected:
Even if I didn't clicked `search`, the immediate change of `date` `select` would not trigger searching until I've clicked `search`

Is there solution to this ?

Maybe Expected Feature:
```ts
export const StateSubmittedParams = atom({
  key: "StateSubmittedParams",
  default: selector({
    key: "StateSubmittedParams/Default",
    get({ get }) {
      const date = get(StateDate);
      const select = get(StateSelect);
      return {
        date,
        select
      };
    }
  }).toPromise()
});


```

**Comment by drarmstr** [comment, marked as answer]
If you want the search to only be initiated from actions instead of staying in sync with all upstream atom changes then you could split your parameter state to separate atoms.  A "pending" atom that is updated as you edit and then transferred to a "submitted" when the search button is clicked. 

It looks like you're mostly doing this.  However, the "submitted" atom shouldn't get the default parameters at all if the intention is to avoid any searches until the button is clicked.
